app/src/main/java/gov/hhs/cdc/trustedintermediary/external/reportstream/ReportStreamLabOrderSender.java

```diff
@@ -147,13 +149,13 @@ protected String requestToken() throws UnableToSendLabOrderException {
     protected String retrievePrivateKey() throws SecretRetrievalException {
         var senderPrivateKey =
                 "report-stream-sender-private-key-" + ApplicationContext.getEnvironment();
-        String key = getCachedPrivateKey();
+        String key = this.keyCache.get(senderPrivateKey);
         if (key != null) {
             return key;
         }
 
         key = secrets.getKey(senderPrivateKey);
-        setCachedPrivateKey(key);
+        this.keyCache.put(senderPrivateKey, key);
```

Comment:
With this change, you can now delete the `getCachedPrivateKey` method, `setCachedPrivateKey` method, and the `cachedPrivateKey` field in this class.

Reply:
Great find.